pex/pex_bootstrapper.py

```diff
@@ -133,7 +133,9 @@ def maybe_reexec_pex(compatibility_constraints):
   Python interpreter to re-exec this pex with.
 
   """
-  if ENV.SHOULD_EXIT_BOOTSTRAP_REEXEC:
+  if os.environ.get('SHOULD_EXIT_BOOTSTRAP_REEXEC'):
```

Comment:
should be able to `os.environ.pop('XX', None)` to retrieve + del in a single call here.